ENIGMAsystem/SHELL/Platforms/SDL/Event.cpp

```diff
@@ -211,9 +212,22 @@ void SDL_Event_Handler::windowResized(const SDL_Event *event) {
   enigma_user::draw_clear(enigma_user::window_get_color());
 }
 
-void SDL_Event_Handler::joyDeviceAdded(const SDL_Event *event) { addGamepad(event->cdevice.which); }
+void SDL_Event_Handler::joyDeviceAdded(const SDL_Event *event) {
+  addGamepad(event->cdevice.which);
+  joysticks.resize(SDL_NumJoysticks(), nullptr);
+  joysticks_open();
+}
 
-void SDL_Event_Handler::joyDeviceRemoved(const SDL_Event *event) { removeGamepad(event->cdevice.which); }
+void SDL_Event_Handler::joyDeviceRemoved(const SDL_Event *event) {
+  removeGamepad(event->cdevice.which);
+  int joystick_count = SDL_NumJoysticks();
+  if (joystick_count < 0) {
+    joysticks.resize(0);
+  } else {
+    joysticks.resize(joystick_count);
```

Comment:
I don't think you want to be shrinking the vector at all. This will cause issue say if you have 3 joysticks and you remove the 2nd one. You then would have joystick {js1, nullptr, js2} and you would shrink the vector to {js1, nullptr} is not what you want. On add you should iterate the vector to check for the first nullptr and then resize the vector if you need to add additional joysticks.